src/ext/JourneyPatternStopPointMap/JourneyPatternStopPointMap.tsx

```diff
@@ -38,27 +74,88 @@ export const JourneyPatternStopPointMap = ({
       stopPointLocationSequence: [],
       showQuaysState: {},
       hideNonSelectedQuaysState: {},
+      focusedMarker: undefined,
     },
   );
+  const mapStateRef = useRef<MapState>({
+    quayStopPointSequenceIndexes: {},
+    stopPointLocationSequence: [],
+    showQuaysState: {},
+    hideNonSelectedQuaysState: {},
+    focusedMarker: undefined,
+  });
 
-  const stopPlaces = useAppSelector((state) => state.stopPlaces)?.stopPlaces;
-  const quayLocationsIndex = useAppSelector(
+  const stopPlacesState: StopPlacesState = useAppSelector(
     (state) => state.stopPlaces,
-  )?.quayLocationsIndex;
-  const quayStopPlaceIndex = useAppSelector(
-    (state) => state.stopPlaces,
-  )?.quayStopPlaceIndex;
+  ) as StopPlacesState;
+  const stopPlaces = stopPlacesState?.stopPlaces || [];
+  const searchedStopPlaces = stopPlacesState?.searchedStopPlaces;
+  const totalStopPlaces = useMemo(() => {
+    const total = stopPlaces ? [...stopPlaces] : [];
+    searchedStopPlaces?.forEach((stopPlace) => {
+      if (stopPlaces.filter((s) => s.id === stopPlace.id).length === 0) {
+        total.push(stopPlace);
+      }
+    });
+    return total;
+  }, [stopPlaces, searchedStopPlaces]);
```

Comment:
This still has the same problem. See line 35 where stopPlaces is set to the empty array as a fallback mechanism. Since useEffect uses reference equality for objects (arrays are objects) and `[] !== []` this will cause the useEffect to run on each render as long as the conditional falls through.